### PR TITLE
fix: added self-contained styles for alert component

### LINKED ISSUE #97: Self-contained styles for Alert
similar as https://github.com/SAP/fundamental-styles/pull/60

### PR DESCRIPTION
**Pull request by SaiShan**
Closes #97 

This PR contains the self-contained styling for the alert component.
This PR is part of the larger task of making all components self-contained #136

### CODE REVIEW

File: components/alert.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+$fd-icons-path: "../scss/icons/";
+$fd-fonts-path: "../scss/fonts/";
+
+@import "../scss/components/alert";
+@import "../scss/components/link";
+@import "../scss/core/elements";
+@import "../scss/core/root";
```
Comment (greg-a-smith) [inline components/alert.scss]: Importing the entire core elements and root files is not what we want to do here. We need to find out what from these files is necessary to render alerts properly and bring those into the link class definitions -- either via common mixins or directly. The whole point of self-contained styles is not only bring along what is needed and not do the changes using elements.

File: scss/components/alert.scss [hunk 2]
```diff
@@ -161,3 +161,8 @@ $block: #{$fd-namespace}-alert;
       }
     }
 }
+
+p {
+  margin-bottom: 0;
+  margin-top: 0;
+}
```
Comment (greg-a-smith) [inline scss/components/alert.scss]: Styles cannot be applied directly to elements (`<p>` in this case) as they can bleed into other components on the page.

File: scss/mixins/_mixins.scss [hunk 3]
```diff
@@ -239,10 +242,18 @@
...
   @include fd-disabled() {
     @include fd-var-color("color", map-get($fd-colors-action-states, "disabled"), --fd-color-action-disabled);
     @include fd-var-color("border-bottom-color", map-get($fd-colors-action-states, "disabled"), --fd-color-action-disabled);
+    @include fd-var-color("outline-color", fd-color-state("disabled", "action"), --fd-color-action-disabled);
+    cursor: not-allowed;
+  }
+  &:focus {
+    outline-style: dotted;
+    outline-width: 1px;
+    @include fd-var-color("outline-color", fd-color-state("hover", "action"), --fd-color-action-focus);
```
Comment (greg-a-smith) [inline scss/mixins/_mixins.scss]: If all the changes in this file are needed, I would expect these to be part of the link component's PR.  Once they are there, this PR can take advantage of those same styles.

File: scss/components/alert.scss (hunk 2 above)
Comment (SaiShan) [reply scss/components/alert.scss]: @greg-a-smith Thank you for the review, will create a class for it to prevent the bleeding.

**Comment by SaiShan** [comment]
Do not review this PR just yet, might have another fix.

**Pull request merged**